yjit/src/codegen.rs

```diff
@@ -4702,6 +4702,32 @@ fn jit_rb_str_bytesize(
     true
 }
 
+fn jit_rb_str_getbyte(
+    _jit: &mut JITState,
+    asm: &mut Assembler,
+    _ocb: &mut OutlinedCb,
+    _ci: *const rb_callinfo,
+    _cme: *const rb_callable_method_entry_t,
+    _block: Option<BlockHandler>,
+    _argc: i32,
+    _known_recv_class: *const VALUE,
+) -> bool {
+    asm.comment("String#getbyte");
+    extern "C" {
+        fn rb_str_getbyte(str: VALUE, index: VALUE) -> VALUE;
+    }
+    // rb_str_getbyte() never allocates
```

Comment:
@tenderlove pointed out that though this doesn't allocate it (likely) does raise on non-integers so I think we might need to jit_prepare_routine_call after all. Maybe a test would be good to add too that it raises properly when the argument is a non-integer.

(sorry I didn't notice earlier 😅)

Reply:
FYI this would change the behavior of a backtrace when it does get raised because we don't push a frame for `MethodGenFn`. It's fine for methods that have a corresponding `opt_*` insn since the interpreter breaks it too, but it's not the case for this method. We already seem to have `String#+@` as another such example (that doesn't have `opt_` insn), so it's not new in this PR, but we should still be careful not to be too aggressive about this.

Reply:
Ah, right. Maybe we should consider a fixnum guard instead?

Reply:
If it is easy enough and still helps the benchmarks you want to improve, that'd be great. But I'm personally open to merging it as is as long as there are compelling benchmark improvements.